mulog-prometheus/src/com/brunobonacci/mulog/publishers/prometheus.clj

```diff
@@ -41,31 +49,99 @@
           (publish-records! config (transform (map second items)))
           (rb/dequeue buffer last-offset)))))
 
+  com.brunobonacci.mulog.publishers.prometheus.registry.ReadRegistry
+  (get-registry [_]
+    (reg/get-registry registry))
 
-  java.io.Closeable
-  (close [_]
-    ;; TODO:
-    ))
+  (write-out [_ out]
+    (reg/write-out registry out))
 
+  (write-text [_]
+    (reg/write-text registry))
 
+  java.io.Closeable
+  (close [_]))
 
-(def ^:const DEFAULT-CONFIG
+(def ^:private DEFAULT-CONFIG
   {:max-items     1000
    :publish-delay 100
-   ;; publish-method :push-gateway | :scrape
-   ;; push-gateway-endpoint ""
-   ;; :register    (...)
-   ;; ....
-
-   ;; function to transform records
-   :transform     identity
-   })
-
 
+   ;; You can supply your own registry which will be used for all events.
+   ;; If you do not specify a registry, the default registry is used.
+   ;; The default registry is the static `CollectorRegistry.defaultRegistry`
+   :registry      (reg/create-default)
+
+   ;; You can setup the prometheus-publisher to push to a prometheus push gateway.
+   ;; `job` is always required - These are what your metrics are identified by.
+   ;; Typically a unique job name or application name.
+   ;;
+   ;; You can then either supply your own `^PushGateway :gateway` or supply an
+   ;; `:endpoint` which will create a new `PushGateway`.
+   :push-gateway  {:endpoint nil
+                   :job      nil
+                   :gateway  nil}
+
+   ;; A function to apply to the sequence of events before publishing.
+   ;; This transformation function can be used to filter, transform,
+   ;; anonymise events before they are published to a external system.
+   ;; by defatult there is no transformation.  (since v0.1.8)
+   :transform         identity
+
+   ;; A function to apply to the sequence of metrics before converting into a collection.
+   ;; You will receive a `com.brunobonacci.mulog.publishers.prometheus.metrics-spec/metric` and
```

Comment:
the function `transform` is applied to the entire buffer of events not just the single events.
this allows the user to filter events that isn't interested.

the function `transform-metrics` does the same. It takes a sequence of metrics and potentially updated sequence of metrics, so the description is incorrect.

Additionally if the intention is to keep spec internal for the moment, then we shouldn't mention it as part of the public api.

Reply:
I have updated the documentation accordingly. Please let me know any other feedback